src/users/users.service.ts

```diff
@@ -104,4 +79,11 @@ export class UsersService {
   async getUserTasks(id: string) {
     await this.userModule.find().populate('tasks', null, Tasks.name).exec();
   }
+
+  async publicUser(email: string) {
+    return this.userModule.findOne({
+      where: { email },
+      attributes: { exclude: ['password'] },
```

Comment:
it's better to add changes to entity and after that fill free to delete publicUser function
  // This column will not be selected by find functions and  QueryBuilder 
  @Column({select: false})